### DISCUSSION TITLE
Does ADetailer work with LORAs in the main prompt?

### DISCUSSION THREAD

**Discussion by andupotorac** [Bing-su/adetailer#136] [General]
I'm trying to use ADetailer to replace/fix faces to the main prompt, but if I have any LORAs applied, it does not seem to work. I also decreased the Detection model confidence threshold to 0.2 to be sure it can detect them, but luck.

I read here, but I wonder if it's only about applying LORAs to images generated in the main prompt without a LORA https://github.com/Bing-su/adetailer/issues/30

Even though I have ADetailer enabled, when I try to use it, it simply does not do it.
`[-] ADetailer: nothing detected on image 2 with 1st settings.`

![image](https://github.com/Bing-su/adetailer/assets/3269738/f3a188d1-f7a1-42f5-a783-edcfa38d790a)

**Comment by andupotorac** [comment]
If I go manually to inpaint over the face, and copy/paste the same prompt from ADetailer, it works. So most likely the issue is with the recognition software. 
<img width="646" alt="image" src="https://github.com/Bing-su/adetailer/assets/3269738/0158fb10-b4f4-41b1-9cf1-941f6e7b9863">

**Comment by andupotorac** [comment]
Wondering if this is related to https://github.com/Bing-su/adetailer#model and which one might fare better.

**Comment by Bing-su** [comment]
yes, it is related to the models. since I used the same dataset, they should be similar.

**Reply by andupotorac** [reply]
Right. So then none of the models will work to detect cartoon faces. Is it possible for me to add face detection models there in the directory if I find any, and select them in the UI in the dropdown?

**Comment by andupotorac** [comment]
@Bing-su Wanted to follow-up on this and CONFIRM it does work with LORAs, even without realistic faces. (hands too when visible)
<img width="2098" alt="Adetailer works with LORAs" src="https://github.com/Bing-su/adetailer/assets/3269738/630db37b-2edd-4f6f-9405-f1088a326ceb">
<img width="628" alt="image" src="https://github.com/Bing-su/adetailer/assets/3269738/d5b79818-9cab-4c71-81a8-b762daa8f653">

**Comment by Bing-su** [comment]
If you leave ad_prompt blank, the original prompt is used, which also includes loras.

**Reply by andupotorac** [reply]
I meant you can actually put the LORA in the ad_prompt, seems to work.
